src/ipfs/uploadIPFSContent.ts

```diff
@@ -0,0 +1,140 @@
+import axios from 'axios';
+
+import { IPFSContentTypes, ContentType, IPFSContentAndType, IPFSUploadServiceResponse } from '../types/ipfs';
+
+// TODO update when BE service is ready
+const uploadToIPFS = async (data: IPFSContentTypes) => {
+  const ipfsURL = process.env.IPFS_GATEWAY_URL;
+
+  if (!ipfsURL) {
+    throw new Error('IPFS_GATEWAY_URL is not set');
+  }
+
+  // POST data to BE service
+  const ipfsHash = await axios
+    .post<IPFSUploadServiceResponse>(ipfsURL, data)
+    .then(response => response.data.ipfsHash)
+    .catch(error => {
+      console.error('Error:', error);
+      throw new Error('Failed to upload data to IPFS');
+    });
+
+  return ipfsHash;
+};
```

Comment:
would it be better for this to take a URL as a param?

```
async function getQuoteAndBuyCoverInputs(
  productId: Integer,
  coverAmount: IntString,
  coverPeriod: Integer,
  coverAsset: CoverAsset,
  coverBuyerAddress: Address,
  slippage: number = DEFAULT_SLIPPAGE / SLIPPAGE_DENOMINATOR,
  ipfsCidOrContent: string | IPFSContentForProductType[ProductTypes] = '',
  coverRouterUrl = process.env.COVER_ROUTER_URL!, <----- need to add a new URL param for IPFS after this
): 
```

Reply:
we replaced coverRouterUrl with a `nexusApiUrl` that will be used for both cover-router and IPFS API. it also has a default value so the user does not to worry about it, unless they want to change the API URL.